crates/wasmtime/src/signatures.rs

```diff
@@ -259,4 +250,18 @@ impl SignatureRegistry {
             .get(index.bits() as usize)
             .and_then(|e| e.as_ref().map(|e| &e.ty).cloned())
     }
+
+    /// Registers a single function with the collection.
+    ///
+    /// Returns the shared signature index for the function.
+    pub fn register(&self, ty: &WasmFuncType) -> VMSharedSignatureIndex {
+        self.0.write().unwrap().register(ty)
+    }
+
+    /// Registers a single function with the collection.
+    ///
+    /// Returns the shared signature index for the function.
+    pub unsafe fn unregister(&self, sig: VMSharedSignatureIndex) {
+        self.0.write().unwrap().unregister_entry(sig, 1)
+    }
```

Comment:
These are only used for `HostFunc` and correspond to the creation of a `HostFunc` and its deletion.